c++/src/kj/common-test.c++

```diff
@@ -75,6 +75,37 @@ TEST(Common, Maybe) {
       ADD_FAILURE();
     }
     EXPECT_EQ(123, m.orDefault(456));
+    bool ranLazy = false;
+    EXPECT_EQ(123, m.orDefault([&] {
+      ranLazy = true;
+      return 456;
+    }));
+    EXPECT_FALSE(ranLazy);
+
+    KJ_IF_MAYBE(v, m) {
+      const int& ref = m.orDefault([notUsed = 5]() -> int& { return notUsed; });
```

Comment:
Compiler error because the lambda isn't mutable, but it's also sorta weird that you're returning a reference to a capture which will go away immediately when the lambda is destroyed. Maybe return a reference to a local variable on the stack instead?